### PR TITLE
Fix Flow types of useEffectEvent

### PR DESCRIPTION
**Pull request by eps1lon**
## Summary

Just copied the types over from the internal types. Type error was hidden by overly broad FlowFixMe. With `$FlowFixMe[not-a-function]` we would've seen the actual issue: 
```
Cannot return `dispatcher.useEffectEvent(...)` because  `T` [1] is incompatible with  undefined [2].Flow(incompatible-return)
```

## How did you test this change?

- [x] yarn flow dom-node
- [x] CI

### CODE REVIEW

File: packages/react/src/ReactHooks.js [hunk 1]
```diff
@@ -234,8 +234,10 @@ export function useMemoCache(size: number): Array<any> {
   return dispatcher.useMemoCache(size);
 }
 
-export function useEffectEvent<T>(callback: T): void {
+export function useEffectEvent<Args, Return, F: (...Array<Args>) => Return>(
```
Comment (eps1lon) [inline packages/react/src/ReactHooks.js:237]: @kassens Do we actually need the `Args` and `Return` type parameter here? In TS we can just do `useEffectEvent<T extends Function>`. Does something similar exist in Flow?

File: packages/react/src/ReactHooks.js [hunk 2]
```diff
@@ -218,24 +218,26 @@ export function useSyncExternalStore<T>(
...
 
 export function useMemoCache(size: number): Array<any> {
   const dispatcher = resolveDispatcher();
-  // $FlowFixMe This is unstable, thus optional
+  // $FlowFixMe[not-a-function] This is unstable, thus optional
   return dispatcher.useMemoCache(size);
 }
 
-export function useEffectEvent<T>(callback: T): void {
+export function useEffectEvent<Args, Return, F: (...Array<Args>) => Return>(
```
Comment (kassens) [inline packages/react/src/ReactHooks.js:237]: This is a little bit shorter, we don't have to define the return type in the type constraint for F:
```suggestion
export function useEffectEvent<Args, Return, F: (...Array<Args>) => Return>(
export function useEffectEvent<Args, F: (...Args) => mixed>(
```

**Review by kassens — APPROVED** [review]
Nice, we should probably try to replace all the generic suppressions with specific ones to avoid shadowing new errors of different kinds.

File: packages/react/src/ReactHooks.js (hunk 2 above)
Comment (eps1lon) [reply packages/react/src/ReactHooks.js]: Applied in [facebook/react@`9cf21fb` (#26468)](https://github.com/facebook/react/pull/26468/commits/9cf21fb9f227cedbf6c0dc896fcf32e856d2e296) to make it consistent with the internal types.

**Pull request merged**